pkg/provider/loadBalancer.go

```diff
@@ -238,45 +286,95 @@ func syncLoadBalancer(ctx context.Context, kubeClient kubernetes.Interface, serv
 	return &service.Status.LoadBalancer, nil
 }
 
-func discoverPool(cm *v1.ConfigMap, namespace, configMapName string) (pool string, global bool, err error) {
-	var cidr, ipRange string
+func getConfigWithNamespace(cm *v1.ConfigMap, namespace, name string) (value, key string, err error) {
 	var ok bool
 
-	// Find Cidr
-	cidrKey := fmt.Sprintf("cidr-%s", namespace)
-	// Lookup current namespace
-	if cidr, ok = cm.Data[cidrKey]; !ok {
-		klog.Info(fmt.Errorf("no cidr config for namespace [%s] exists in key [%s] configmap [%s]", namespace, cidrKey, configMapName))
-		// Lookup global cidr configmap data
-		if cidr, ok = cm.Data["cidr-global"]; !ok {
-			klog.Info(fmt.Errorf("no global cidr config exists [cidr-global]"))
+	key = fmt.Sprintf("%s-%s", name, namespace)
+
+	if value, ok = cm.Data[key]; !ok {
+		return "", key, fmt.Errorf("no config for %s", name)
+	}
+
+	return value, key, nil
+}
+
+func getConfig(cm *v1.ConfigMap, namespace, configMapName, name, what, pool string) (value string, global bool, err error) {
+	var key string
+
+	value, key, err = getConfigWithNamespace(cm, namespace, name)
+	if err != nil {
+		klog.Info(fmt.Errorf("no %s config for namespace [%s] exists in key [%s] configmap [%s]", name, namespace, key, configMapName))
+		value, key, err = getConfigWithNamespace(cm, "global", name)
+		if err != nil {
+			klog.Info(fmt.Errorf("no global %s config exists [%s]", name, key))
 		} else {
-			klog.Infof("Taking address from [cidr-global] pool")
-			return cidr, true, nil
+			klog.Infof("Taking %s from [%s]%s", what, key, pool)
```

Comment:
Maybe could use simple version, otherwise it will be little bit too complicated to use
```suggestion
			klog.Infof("Taking %s from [%s]", type, key)
```